src/main/java/com/cloudhopper/smpp/impl/DefaultSmppClient.java

```diff
@@ -273,28 +274,36 @@ protected Channel createConnectedChannel(String host, int port, long connectTime
         // a socket address used to "bind" to the remote system
         InetSocketAddress socketAddr = new InetSocketAddress(host, port);
 
-	// set the timeout
-	this.clientBootstrap.setOption("connectTimeoutMillis", connectTimeoutMillis);
+        // set the timeout
+        this.clientBootstrap.setOption("connectTimeoutMillis", connectTimeoutMillis);
 
         // attempt to connect to the remote system
         ChannelFuture connectFuture = this.clientBootstrap.connect(socketAddr);
-        
-        // wait until the connection is made successfully
-	// boolean timeout = !connectFuture.await(connectTimeoutMillis);
-	// BAD: using .await(timeout)
-	//      see http://netty.io/3.9/api/org/jboss/netty/channel/ChannelFuture.html
-	connectFuture.awaitUninterruptibly();
-	//assert connectFuture.isDone();
-
-	if (connectFuture.isCancelled()) {
-	    throw new InterruptedException("connectFuture cancelled by user");
-	} else if (!connectFuture.isSuccess()) {
-	    if (connectFuture.getCause() instanceof org.jboss.netty.channel.ConnectTimeoutException) {
-		throw new SmppChannelConnectTimeoutException("Unable to connect to host [" + host + "] and port [" + port + "] within " + connectTimeoutMillis + " ms", connectFuture.getCause());
-	    } else {
-		throw new SmppChannelConnectException("Unable to connect to host [" + host + "] and port [" + port + "]: " + connectFuture.getCause().getMessage(), connectFuture.getCause());
-	    }
-	}
+
+        // Wait until the connection is made successfully.
+        // According to the netty documentation it is bad to use .await(timeout). Instead
+        //     b.setOption("connectTimeoutMillis", 10000);
+        // should be used. See http://netty.io/3.9/api/org/jboss/netty/channel/ChannelFuture.html
+        // It turns out that under certain unknown circumstances the connect waits forever: https://github.com/twitter/cloudhopper-smpp/issues/117
+        // That's why the future is canceled 1 second after the specified timeout.
+        // This is a workaround and hopefully not needed after the switch to netty 4.
+        if (!connectFuture.await(connectTimeoutMillis + 1000)) {
```

Comment:
I originally used two minutes just to be on the safe side, but I think 1 second should do the trick. I don't want to use the same time as the real timeout in order not to interfere with the real timeout.